CMakeLists.txt

```diff
@@ -191,6 +191,36 @@ endif()
 option(WITH_MEMORY_ANALYZER
   "build the memory analyzer" ${WITH_MEMORY_ANALYZER_DEFAULT})
 
+if((CMAKE_MAJOR_VERSION EQUAL 3 AND CMAKE_MINOR_VERSION GREATER_EQUAL 9)
+   AND (CMAKE_BUILD_TYPE STREQUAL "Release"))
+  cmake_policy(SET CMP0069 NEW)
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT ipo_supported OUTPUT error)
+endif()
+
+if(ipo_supported)
+  message(STATUS "IPO / LTO enabled")
+  set_property(GLOBAL PROPERTY INTERPROCEDURAL_OPTIMIZATION True)
+  add_definitions(-DLTO)
+  if ((CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+       OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "i386")
+       OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "i586")
+       OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "i686")
+       OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "amd64") # freebsd
+       OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64") # windows
+       OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86"))
+     set(PROCESSOR_FAMILY "Intel")
+     if(CMAKE_COMPILER_IS_GNUCC
+         OR (CMAKE_C_COMPILER_ID STREQUAL AppleClang)
+         OR (CMAKE_C_COMPILER_ID STREQUAL Clang)
+         OR (CMAKE_C_COMPILER_ID STREQUAL Intel))
+       add_definitions(-march=native)
```

Comment:
```suggestion
     if(CMAKE_COMPILER_IS_GNUCC
         OR (CMAKE_CXX_COMPILER_ID STREQUAL AppleClang)
         OR (CMAKE_CXX_COMPILER_ID STREQUAL Clang)
         OR (CMAKE_CXX_COMPILER_ID STREQUAL Intel))
       set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native")
```

Reply:
CXX compiler! good catch

Reply:
I think you might actually need _both_ the C and CXX variants?